charts/pravega/templates/post-install-upgrade-hooks.yaml

```diff
@@ -0,0 +1,107 @@
+kind: Role
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+rules:
+- apiGroups:
+  - pravega.pravega.io
+  resources:
+  - "*"
+  verbs:
+  - get
+---
+
+kind: RoleBinding
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+subjects:
+- kind: ServiceAccount
+  name: {{ template "hook.fullname" . }}
+  namespace: {{.Release.Namespace}}
+roleRef:
+  kind: Role
+  name: {{ template "hook.fullname" . }}
+  apiGroup: rbac.authorization.k8s.io
+
+---
+
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+    "helm.sh/hook": post-install, post-upgrade
+    "helm.sh/hook-weight": "1"
+    "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+
+---
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: {{ template "hook.fullname" . }}
+  annotations:
+      "helm.sh/hook": post-install, post-upgrade
+      "helm.sh/hook-weight": "1"
+      "helm.sh/hook-delete-policy": hook-succeeded, before-hook-creation, hook-failed
+data:
+  validations.sh: |
+    #!/bin/sh
+    set -e
+    sleep 30
+    kubectl get PravegaCluster -n {{ .Release.Namespace }} {{ template "pravega.fullname" . }} -o jsonpath='{.status.conditions[?(@.type=="PodsReady")].status}' | grep True
+    currentReplicas=`kubectl get PravegaCluster -n {{ .Release.Namespace }} {{ template "pravega.fullname" . }} -o jsonpath='{.status.currentReplicas}'`
+    readyReplicas=`kubectl get PravegaCluster -n {{ .Release.Namespace }} {{ template "pravega.fullname" . }} -o jsonpath='{.status.readyReplicas}'`
```

Comment:
Is it possible to have `status.currentReplicas` equal to `status.readyReplicas` but less than `status.replicas`?
I mean the situation where not all the replicas have been created up start, and the ones that have been created are all in the ready state, while some other replicas have not been created yet?

Reply:
I do agree that it would be safer to have `.status.replicas` instead of `.status.currentReplicas`. I have incorporated that change.